File: examples/test_alloc.f90
      program test
      use Damaris

      type(c_ptr) :: cptr
      integer :: ierr,id
      integer :: i,j,k
      real,  pointer :: mydata(:,:,:)
      
 
      id     = 42 

      call df_initialize("config.xml",id,ierr)

      cptr = df_alloc("my group/my variable",1,ierr)
      call c_f_pointer(cptr,mydata,[64,16,4])
 
      do i =  1, 64
      do j =  1, 16
      do k =  1, 4
         mydata(i,j,k) = i*j*k
      end do
      end do
      end do 
 
      call df_commit("my group/my variable",1,ierr);

      call df_signal("my event",1,ierr);
 
      call df_finalize(ierr)

      end program test

File: examples/test_alloc.c
#include <stdio.h>
#include "include/Damaris.h"      

int main(int argc, char** argv) 
{
	int id = 0;
	float* mydata = NULL;
	int x = 64;
	int y = 16;
	int z = 4;

	char* varname = "my group/my variable";
	char* event   = "my event";

	if(argc != 2) {
		printf("Usage: ./test_alloc_c <config.xml>\n");
		exit(0);
	}

	DC_initialize(argv[1],id);	

	mydata = (float*)DC_alloc(varname,0);

	if(mydata == NULL) {
		printf("Error when allocating buffer\n");
		exit(0);
	}

	int i,j,k;
	for(i = 0; i < x; i++) {
	for(j = 0; j < y; j++) {
	for(k = 0; k < z; k++) {
		mydata[i*y*z+j*z+k] = (float)(i*j*k);
	}
	}
	}
		
	DC_commit(varname,0);
	DC_signal(event,0);
	DC_finalize();

	return 0;
}
